### PR TITLE
Make all models work && add reverse peak model

### PR DESCRIPTION
**Pull request by rozetko**
- Subtract min value from dataset before passing to model
- Rename StepModel -> DropModel
- Use cache to save state in all models
- Return `Segment { 'from': <timestamp>, 'to': <timestamp>}` instead of `Segment { 'from': <index>, 'to': <index>}` in all models
- Integrate new peaks model (from https://github.com/hastic/hastic-server/pull/123)
- Integrate new reverse-peaks model (from https://github.com/hastic/hastic-server/pull/123)
- Refactor: make `predict` method in `Model` not abstract and remove it from all children
- Refactor: add abstract `do_predict` method to models

### CODE REVIEW

File: analytics/analytic_unit_manager.py [hunk 1]
```diff
@@ -56,4 +55,10 @@ def ensure_worker(analytic_unit_id, analytic_unit_type) -> AnalyticUnitWorker:
             'error': str(e)
         }
 
+def prepare_data(data):
```
Comment (jonyrock) [inline analytics/analytic_unit_manager.py]: please add a comment what is it for or a better name

Comment (jonyrock) [reply analytics/analytic_unit_manager.py]: I think is should be a converation. Please also add typings here.

File: analytics/analytic_unit_worker.py [hunk 2]
```diff
@@ -13,8 +14,8 @@ def __init__(self, analytic_unit_id: str, detector: detectors.Detector):
         self.analytic_unit_id = analytic_unit_id
         self.detector = detector
 
-    async def do_learn(self, segments: list, data: pd.DataFrame, cache: dict) -> dict:
+    async def do_learn(self, segments: list, data: pd.DataFrame, cache: Optional[dict]) -> dict:
```
Comment (jonyrock) [inline analytics/analytic_unit_worker.py]: I like it

File: analytics/analytic_unit_worker.py [hunk 3]
```diff
@@ -13,8 +14,8 @@ def __init__(self, analytic_unit_id: str, detector: detectors.Detector):
...
         return await self.detector.train(data, segments, cache)
 
-    async def do_predict(self, data: pd.DataFrame, cache: dict) -> dict:
+    async def do_predict(self, data: pd.DataFrame, cache: Optional[dict]) -> dict:
```
Comment (jonyrock) [inline analytics/analytic_unit_worker.py]: you also can add a type "AnalyticUnitCache = dict" to make this "dict" more friendly

File: analytics/detectors/pattern_detector.py [hunk 4]
```diff
@@ -30,23 +33,23 @@ def __init__(self, pattern_type):
...
+        new_cache = self.model.fit(dataframe, segments, cache)
         return {
-            'cache': cache
+            'cache': new_cache
         }
 
-    async def predict(self, dataframe: pd.DataFrame, cache: dict):
-        predicted = await self.model.predict(dataframe, cache)
+    async def predict(self, dataframe: pd.DataFrame, cache: Optional[dict]):
+        predicted = self.model.predict(dataframe, cache)
```
Comment (jonyrock) [inline analytics/detectors/pattern_detector.py]: in theory we need to split big dataframe to small chunks and do `yield asyncio.sleep(1)`
but please don`t fix it now

Comment (jonyrock) [reply analytics/detectors/pattern_detector.py]: we can just make `#TODO: split and sleep`

File: analytics/models/drop_model.py [hunk 5]
```diff
@@ -76,46 +76,34 @@ def fit(self, dataframe: pd.DataFrame, segments: list, cache: dict) -> dict:
                 convolve_list.append(max(convolve))
 
         if len(confidences) > 0:
-            self.state['confidence'] = min(confidences)
+            self.state['confidence'] = float(min(confidences))
```
Comment (jonyrock) [inline analytics/models/drop_model.py]: just interested why do we need `float` here

Comment (rozetko) [reply analytics/models/drop_model.py]: It is `np.int64` by default and can't be serialized by Python

File: analytics/analytic_unit_manager.py (hunk 1 above)
Comment (rozetko) [reply analytics/analytic_unit_manager.py]: It makes timestamp column `datetime` type
And subtracts min dataset value from dataset

**Pull request merged**